firedrake/mesh.py

```diff
@@ -3768,7 +3768,7 @@ def _parent_mesh_embedding(
         # any rank which will have been set to comm.size + 1)
         owned_ranks_tosort[owned_ranks != parent_mesh.comm.rank] = parent_mesh.comm.size
         # now a sort by rank will give us the ordering we want
-        idxs = np.argsort(owned_ranks_tosort, kind='stable')
+        idxs = np.argsort(owned_ranks_tosort)
```

Comment:
Why have you removed this?

Reply:
Note that this entire block has been removed.

The way `pointSF` was initially constructed depended on the assumption that swarm points were ordered according to the global coordinate indices, which was achieved with `stable`. Given that `plex` points can come in arbitrary order (at least stratum-wise), such underlying assumption should not be necessary.